pkg/sql/sqlbase/structured.go

```diff
@@ -3122,6 +3189,84 @@ func (desc *MutableTableDescriptor) MakeMutationComplete(m DescriptorMutation) e
 					return err
 				}
 			}
+		case *DescriptorMutation_ComputedColumnSwap:
+			args := t.ComputedColumnSwap
+
+			oldCol, err := desc.FindColumnByID(args.OldColumnId)
+			if err != nil {
+				return err
+			}
+			newCol, err := desc.FindColumnByID(args.NewColumnId)
+			if err != nil {
+				return err
+			}
+
+			newCol.ComputeExpr = nil
+
+			unresolvedName := func(name string) *tree.UnresolvedName {
+				return &tree.UnresolvedName{
+					NumParts: 1,
+					Parts:    tree.NameParts{name},
+				}
+			}
+
+			// oldCol should still have new values written to newCol in case
+			// it has a read from previous version.
+
+			oldColComputeExpr := tree.CastExpr{Expr: unresolvedName(oldCol.Name), Type: oldCol.DatumType(), SyntaxMode: tree.CastShort}
+			s := tree.Serialize(&oldColComputeExpr)
+			oldCol.ComputeExpr = &s
+
+			//Rename newCol to oldCol.Name.
+			//Generate unique name for old column.
+			nameExists := func(name string) bool {
+				_, _, err := desc.FindColumnByName(tree.Name(name))
+				return err == nil
+			}
+
+			uniqueName := GenerateUniqueConstraintName(
+				newCol.Name,
+				nameExists,
+			)
+
+			oldColName := oldCol.Name
+			desc.RenameColumnDescriptor(oldCol, uniqueName)
+			desc.RenameColumnDescriptor(newCol, oldColName)
+
+			// Swap Column Family references.
+			// This preserves the ordering of column families when querying for it.
+			for i := range desc.Families {
+				for j := range desc.Families[i].ColumnIDs {
+					if desc.Families[i].ColumnIDs[j] == oldCol.ID {
+						desc.Families[i].ColumnIDs[j] = newCol.ID
+						desc.Families[i].ColumnNames[j] = newCol.Name
+					} else if desc.Families[i].ColumnIDs[j] == newCol.ID {
+						desc.Families[i].ColumnIDs[j] = oldCol.ID
+						desc.Families[i].ColumnNames[j] = oldCol.Name
+					}
+				}
+			}
+
+			// Swap Column Orders.
+			*newCol, *oldCol = *oldCol, *newCol
+
+			// After the swap, newCol points to the oldCol and vice versa.
+			newCol, oldCol = oldCol, newCol
+			newCol.LogicalColumnID = oldCol.LogicalColumnID
+			oldCol.LogicalColumnID = 0
+
+			oldCol.AlterColumnTypeDropColumn = true
+
+			//Drop old column.
+			for i := range desc.Columns {
+				if desc.Columns[i].ID == oldCol.ID {
+					desc.AddColumnMutation(oldCol, DescriptorMutation_DROP)
+					// Use [:i:i] to prevent reuse of existing slice, or outstanding refs
```

Comment:
Do you need to do this? What goes wrong when you dont?

Reply:
I'm not sure actually, I took this from the drop column code.